gemonlinedb/include/gem/onlinedb/ConfigurationProvider.h

```diff
@@ -0,0 +1,76 @@
+#ifndef GEM_ONLINEDB_CONFIGURATIONPROVIDER_H
+#define GEM_ONLINEDB_CONFIGURATIONPROVIDER_H
+
+#include "gem/onlinedb/AMC13Configuration.h"
+#include "gem/onlinedb/AMCConfiguration.h"
+#include "gem/onlinedb/OHv3Configuration.h"
+#include "gem/onlinedb/VFAT3ChipConfiguration.h"
+
+#include "gem/onlinedb/ConfigurationTraits.h"
+#include "gem/onlinedb/PartReference.h"
+
+namespace gem {
+    namespace onlinedb {
+
+        /**
+         * @brief Interface that provides configurations for parts.
+         *
+         * Objects implementing this interface can be used to retrieve the
+         * configuration for parts of a GEM system. It is primarily intended for
+         * use with @ref ConfigurationLinker.
+         *
+         * Retrieval functions throw @c std::out_of_range when the requested
+         * configuration is not found.
+         */
+        class ConfigurationProvider
+        {
+        public:
+            /// @brief Constructor.
+            explicit ConfigurationProvider() = default;
```

Comment:
What's the purpose of declaring default constructor with no args as `explicit`?

Reply:
I generally mark *all* constructors as explicit. It gives a nice symmetry with the `virtual` destructors.